### ISSUE TITLE
Incorrect to handle the block color inversion

### ISSUE THREAD

**Issue by gengshenghong** [jmoenig/Snap--Build-Your-Own-Blocks#1620]
See below picture for example, when two blocks with same color combined, snap will use inverse color (which is really nice feature), but see the right one in below picture. Know how it happened?

Step to get picture 1 result:
1. prepare an empty "forever" block
2. drag "move" block into "forever"
3. drag "wait" block at the bottom of "move" inside "forever"
Step to get picture 2 result:
1. prepare an empty "forever" block
2. drag "move" block at the script blank area
3. drag "wait" block at the bottom of "move" to combine them together (now, it is "move + wait" block)
4. drag "move + wait" block into "forever" block, now check the result! :)

![image](https://cloud.githubusercontent.com/assets/1018468/21975104/acc4efd4-dc05-11e6-925d-5f14f4c16c92.png

**Comment by gengshenghong** [comment]
by the way, reproduce with online version:
http://snap.berkeley.edu/snapsource/snap.html

**Comment by brianharvey** [comment]
Yeah, apparently only the first block of the script being dragged into the slot is eligible for zebra coloring.  Good catch.

**Comment by jmoenig** [comment]
Yeah, there are many of those little zebra-coloring issues. I've "fixed" the case where the new version auto-wraps the first C-slot around a stack of command blocks to also fix the zebra coloring, but some cases still remain.

**Comment by gengshenghong** [comment]
Great to know it's called "zebra-coloring". :)
I think it is very common to operate like my 2nd case in fact, we write some part of "code" first, and put it into another slot block.
